src/Native/System.Native/pal_networkstatistics.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_config.h"
+
+// These functions are only used for platforms which support
+// using sysctl to gather protocol statistics information.
+// Currently, this is all keyed off of whether the include tcp_var.h
+// exists, but we may want to make this more granular for differnet platforms.
+
+#if HAVE_TCP_VAR_H
+
+#include "pal_utilities.h"
+#include "pal_networkstatistics.h"
+
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <sys/socketvar.h>
+#include <netinet/ip.h>
+#include <netinet/ip_var.h>
+#include <netinet/tcp.h>
+#include <netinet/tcp_var.h>
+
+extern "C" void GetTcpGlobalStatistics(TcpGlobalStatistics* retStats)
+{
+    memset(retStats, 0, sizeof(TcpGlobalStatistics));
```

Comment:
Good call, I'll add the assertion.  I may also just remove the memset altogether. I previously thought there might be unused fields in the structure, but I tore all of the extra fields out that I wasn't able to fill from the OS data, so the only ones left are ones that I assign to.